modules/java/generator/src/cpp/Mat.cpp

```diff
@@ -1919,12 +1919,12 @@ template<class ARRAY> static jint java_mat_put(JNIEnv* env, jlong self, jint row
 extern "C" {
 
 JNIEXPORT jint JNICALL Java_org_opencv_core_Mat_nPutB
-    (JNIEnv* env, jclass, jlong self, jint row, jint col, jint count, jbyteArray vals);
+    (JNIEnv* env, jclass, jlong self, jint row, jint col, jint count, jint offset, jbyteArray vals);
```

Comment:
This original interface should be restored too (it is JNIEXPORT , not internal).
Please add tests to ensure that this really works and these changes will not be broken in the future (somewhere [here](https://github.com/opencv/opencv/blob/3.4.0/modules/core/misc/java/test/MatTest.java))